### DISCUSSION TITLE
Use shell variable instead of environment variable

### DISCUSSION THREAD

**Discussion by sergiomiranda** [JanDeDobbeleer/oh-my-posh#2776] [Troubleshoot] [answered]
In my theme I have the following code for the transient prompt:

```
  "transient_prompt": {
    "template": "<p:dark-gray,transparent>{{ repeat (.Env.COLUMNS | int) \"_\" }}</>\n- {{ if .Root }}<p:light-red,transparent>\uf0e7 </>{{ end }}<p:light-red,transparent>{{ .PWD }} </>\ue602",
    "foreground": "p:light-gray",
    "background": "transparent"
  },
```

For better readability, this prompt draws a line with the "_" character at full width of the terminal.
To know how many characters to draw, I am making use of a $COLUMNS environment variable that I set from $Host.UI.RawUI.WindowSize.Width with the function Set-EnvVar in PowerShell, as suggested in the documentation.
It works great in PowerShell.

![image](https://user-images.githubusercontent.com/497370/189782918-c5acbcdd-cba2-4c75-8c42-e70c00b7d1a7.png)

In zsh however, I can't find a way to replicate this. The $COLUMNS shell variable already exists as set by the shell itself, but I can't find any way to use it or know of any way to set it as an environment variable like in PowerShell.

1. Is it possible to use a shell variable in templates like environment variables?
2. Is there any way to use in zsh a function like Set-EnvVar in PowerShell to set environment variables?

Thanks!

**Comment by JanDeDobbeleer** [comment, marked as answer]
@sergiomiranda let me check if we can override functions easily in zsh ;-) There's no way to reference shell variables from oh-my-posh.

**Reply by sergiomiranda** [reply]
Hi! Did you have the chance to take a look at this?
Just to know if I there is still hope. Thanks!

**Reply by JanDeDobbeleer** [reply]
It's already [live](https://ohmyposh.dev/docs/configuration/templates#environment-variables).

**Reply by sergiomiranda** [reply]
Thanks, it was exactly what I needed!
